src/language-server/engine/index.ts

```diff
@@ -56,26 +55,15 @@ export class ApolloEngineClient {
           console.log(`[Network Error] ${response.bodyText}`);
         }
       }),
-      setContext((_, request) => {
-        if (!request.headers) request.headers = {};
-        request.headers["x-api-key"] = this.engineKey;
-        if (this.clientIdentity && this.clientIdentity.name) {
-          request.headers["apollo-client-name"] = this.clientIdentity.name;
-          request.headers["apollo-client-reference-id"] =
-            this.clientIdentity.referenceID;
-          request.headers["apollo-client-version"] =
-            this.clientIdentity.version;
-        } else {
-          // default values
-          request.headers["apollo-client-name"] = "Apollo Language Server";
-          request.headers["apollo-client-reference-id"] =
-            "146d29c0-912c-46d3-b686-920e52586be6";
-          request.headers["apollo-client-version"] =
-            require("../../package.json").version;
-        }
-        return request;
+      createHttpLink({
+        uri: baseURL,
+        headers: {
+          ["x-api-key"]: this.engineKey,
+          ["apollo-client-name"]: this.clientIdentity.name,
+          ["apollo-client-reference-id"]: this.clientIdentity.referenceID,
+          ["apollo-client-version"]: this.clientIdentity.version,
+        },
       }),
```

Comment:
This is a slight behavioural change - while before, configuring `clientIdentiy.name` would change the whole code path towards "custom options", now default options for `referenceId` and `version` would still stay in place.

I don't think this was really intended and I'm also not sure if this was ever meant to be configurable for the VSCode extension, so I think this change is okay.

Reply:
That seems reasonable - maybe add that to its own changeset on this PR so it gets called out in release notes?

Reply:
The behaviour itself was never documented to start with, so I'm secretly hoping this one dies a silent death without ever being used 😓